src/common/literal-syntax.k

```diff
@@ -0,0 +1,50 @@
+require "lexical-syntax.k"
+require "list-syntax.k"
+module LITERAL-SYNTAX
+imports LEXICAL-SYNTAX
+imports LIST-SYNTAX
+//@ \subsection{BooleanLiterals.sdf}
+syntax BoolLiteral ::= Bool [klabel('Bool)]
+syntax Bool ::= "true"  [onlyLabel, klabel('True)]
+syntax Bool ::= "false" [onlyLabel, klabel('False)]
+
+//@ \subsection{CharacterLiterals.sdf} todo
+syntax SingleChar// ::= Token{[\r\n\'\\]}//don't understand ^?
+syntax CharContent ::= SingleChar [klabel('Single)]
+                     | UnicodeEscape
+                     | EscapeSeq
+//Problem: Multiple lookahead not fully supported (does not appear when tested in small language)
+syntax CharLiteral //::= "'" CharContent "'"  [klabel('Char)]
+
+//@ \subsection{EscapeSequences.sdf}
+
+syntax EscapeSeq ::= OctaEscape | NamedEscape
+syntax NamedEscape ::= Token{"\\" [btnfr\"\'\\]}         [onlyLabel, klabel('NamedEscape)]
+syntax OctaEscape ::= Token{"\\" [0-7]}                  [onlyLabel, klabel('OctaEscape1)]
+                    | Token{"\\" [0-3][0-7]}             [onlyLabel, klabel('OctaEscape2)]
+                    | Token{"\\" [4-7][0-7]}             [onlyLabel, klabel('OctaEscape2)]
+                    | Token{"\\" [0-3][0-7][0-7]}        [onlyLabel, klabel('OctaEscape3)]
+
+//@ \subsection{FloatingPointLiterals.sdf} todo
+syntax FloatLiteral ::= Float      [klabel('Float)]
+
+//@ \subsection{IntegerLiterals.sdf}
+syntax DeciLiteral ::= Token{[1-9][0-9]* [lL]?} [onlyLabel]
```

Comment:
For integers, sdf defines another sort, like DeciNumeral in between such that
 [1-9][0-9]*  -> DeciNumeral
 DeciNumeral [lL]? -> DeciLiteral

but adding this sort causes errors, and it is not further used, so I just define XXXLiteral

For Float, I just skipped those low level definitions, as long as I put klabel 'Float in the place I want.